Error resend buffer: i/o timeout

New in 2.1.1. Reported by more than one user.

It's a bit unusual as `Write` should seldomly time out. It also times out too soon (less than 1 sec). We set write deadline somewhere?

```
detour: detour.go:249 Error while resend buffer to <masked>:80: write tcp <masked>:49482-><masked>:443: i/o timeout
```

--- a/src/github.com/getlantern/idletiming/idletiming_conn.go
+++ b/src/github.com/getlantern/idletiming/idletiming_conn.go
@@ -202,11 +202,19 @@ func (c *IdleTimingConn) SetDeadline(t time.Time) error {
 }
 
 func (c *IdleTimingConn) SetReadDeadline(t time.Time) error {
+	// zero value means no time out
+	if t.IsZero() {
+		t = epoch
+	}
 	atomic.StoreInt64(&c.readDeadline, t.UnixNano())
 	return nil
 }
 
 func (c *IdleTimingConn) SetWriteDeadline(t time.Time) error {
+	// zero value means no time out
+	if t.IsZero() {
+		t = epoch
+	}
 	atomic.StoreInt64(&c.writeDeadline, t.UnixNano())
 	return nil
 }

